### ISSUE TITLE
Form broken after updating to 2.7.7

### ISSUE THREAD

**Issue by jayesbe** [symfony/symfony#17001]
I recently updated to version 2.7.7 and noticed a form had trouble loading and saving. 

The form uses a dataMapper. In the mapDataToForms method  I am returning an array that looks like this

    array(
        0 => array (
            name => '',
            enabled => ''
        )
        1 => array (
            name => '',
            enabled => ''
        )
    )

I created my ChoiceList object that implements the ChoiceListInterface.  In the method getValuesForChoices(), I NOW have to do this:

    if (count($choices) == 1) { $choices = array_shift($choices); }

I did not have to do that before.
it seems as though if choices contains only one data element, its not the structure as I expect.
its actually contained within another array with a 0 index and ONLY .. when there is one data element.

I downgraded Symfony to 2.7.6 and my form is working as expected. It breaks on 2.7.7. 2.7.5 also working.

**Comment by stof** [comment]
There is no change related to the choice list in 2.7.7. the only changes in the Form component are related to the ``empty_value`` BC layer

**Comment by jayesbe** [comment]
Ok I think I figured it out.

the difference is between a multiple being either true or false.  When true, the form works. When false, there is an error thrown. 

{{{
Notice: Array to string conversion
500 Internal Server Error - ContextErrorException
}}}

Its because Im treating the value as an array anyways. When you make it multiple false it now wants only one value. Maybe this was a bug that was fixed. 

My fix is to just set multiple = true
